src/test/scala/spark/dicom/deidentifier/TestDicomDeidentifier.scala

```diff
@@ -0,0 +1,202 @@
+
+trait WithSpark {
+  var spark = {
+    val spark = SparkSession.builder.master("local").getOrCreate
+    spark.sparkContext.setLogLevel(Level.ERROR.toString())
+    spark
+  }
+}
+
+object TestDicomDeidentifier {
+  val SOME_DICOM_FILEPATH =
+    "src/test/resources/Pancreatic-CT-CBCT-SEG/Pancreas-CT-CB_001/07-06-2012-NA-PANCREAS-59677/201.000000-PANCREAS DI iDose 3-97846/1-001.dcm"
+  lazy val SOME_DICOM_FILE = {
+    val file = new File(SOME_DICOM_FILEPATH)
+    assert(file.exists)
+    file
+  }
+}
+
+class TestDicomDeidentifier
+    extends AnyFunSpec
+    with BeforeAndAfterAll
+    with CancelAfterFailure
+    with WithSpark {
+  import TestDicomDeidentifier._
+
+  val logger = {
+    val logger = LogManager.getLogger(getClass.getName);
+    logger.setLevel(Level.DEBUG)
+    logger
+  }
+
+  override protected def afterAll(): Unit = {
+    spark.stop
+  }
+
+  val SOME_DROPPED_COL = keywordOf(Tag.ContainerComponentID)
+  val SOME_DUMMY_DATE_COL = keywordOf(Tag.ContentDate)
+  val SOME_DUMMY_TIME_COL = keywordOf(Tag.ContentTime)
+  val SOME_DUMMY_DATE_TIME_COL = keywordOf(Tag.AttributeModificationDateTime)
+  val SOME_DUMMY_STRING_COL = keywordOf(Tag.AnnotationGroupLabel)
+  val SOME_ZERO_INT_COL = keywordOf(Tag.PregnancyStatus)
+  val SOME_ZERO_STRING_COL = keywordOf(Tag.ConceptualVolumeDescription)
+  val SOME_EMPTY_STRING_COL = keywordOf(Tag.AccessionNumber)
+  val SOME_CLEANED_COL = keywordOf(Tag.AcquisitionComments)
+
+  describe("Spark") {
+    it("Deidentifies DICOM dataframe using basic profile") {
+      var df = spark.read
+        .format("dicomFile")
+        .load(SOME_DICOM_FILEPATH)
+
+      df = deidentify(df)
+
+      val row = df.first
+
+      assert(
+        row.getAs[String](SOME_DUMMY_DATE_COL)
+          === DUMMY_DATE
+      )
+      assert(
+        row.getAs[String](SOME_DUMMY_TIME_COL)
+          === DUMMY_TIME
+      )
+      assert(
+        row.getAs[String](SOME_DUMMY_DATE_TIME_COL)
+          === DUMMY_DATE_TIME
+      )
+      assert(
+        row.getAs[String](SOME_DUMMY_STRING_COL)
+          === DUMMY_STRING
+      )
+      assert(
+        row.getAs[String](SOME_EMPTY_STRING_COL)
+          === EMPTY_STRING
+      )
+      assertThrows[IllegalArgumentException] {
+        row.fieldIndex(SOME_DROPPED_COL)
+      }
+      // No zero string/int cols in basic profile. To be added later
+      // assert(
+      //   row.getAs[Int](SOME_ZERO_INT_COL)
+      //     === ZERO_INT
+      // )
+      // assert(
+      //   row.getAs[String](SOME_ZERO_STRING_COL)
+      //     === ZERO_STRING
+      // )
+    }
+    it("Deidentifies DICOM dataframe using an option") {
+      var df = spark.read
+        .format("dicomFile")
+        .load(SOME_DICOM_FILEPATH)
+
+      val options = Seq(
+        CleanDesc()
+      )
+
+      df = deidentify(df, options)
+
+      val row = df.first
+
+      assert(
+        row.getAs[String](SOME_CLEANED_COL)
+          === "ToClean"
+      )
+    }
+  }
```

Comment:
It'd be nice to check that option priority is well handled.
It'll allow to "spec" it as well.